File: lista 1 cwiczenia ocaml/zadanie1.ml
let rec flatten xss = 
	if xss = [] then []
	else List.hd xss @ flatten(List.tl xss);; 
let x1 = [[5;6] ; [1;2;3] ; [9;9] ; [22] ];;
flatten x1;;
	
	
	
	
	
	
	(*
	let xs = [ [1;2] ; [3;4]; [5;6] ];;
List.hd xs;;
List.hd(List.tl xs);;
*)

File: lista 1 cwiczenia scala/src/zadanie1.scala
class zadanie1 {

  
  
  def flatten[A](xss: List[ List[A] ]):List[A] =
    if ( xss.isEmpty   ) Nil
    else xss.head ::: flatten( xss.tail )
  
 // val x1 = 5::6::Nil
 // val x2 = 1::2::3::Nil
  
  flatten ( (5::6::Nil) :: (1::2::3::Nil)::Nil )
  
  
  
  
  
  
  
  
  
}